src/Meta/Collection.php

```diff
@@ -48,14 +48,18 @@ public static function get_collection_title( $post_type ) {
 		$post_type_name = 'Blog';
 
 		if ( 'post' !== $post_type ) {
-			$post_type = get_post_type_object( $post_type );
-
-			$post_type_name = $post_type->labels->name;
+			$type = get_post_type_object( $post_type );
+			$post_type_name = $type->labels->name;
 		}
 
-		return apply_filters(
-			sprintf( self::TITLE_FILTER, $post_type ),
-			$post_type_name . ' - ' . get_bloginfo( 'title' )
+		$filter_name = sprintf(
+			self::TITLE_FILTER,
+			is_string( $post_type ) ? $post_type : $post_type_name
+		);
```

Comment:
@mitogh 
I just ran against a bug here and saw your PR
Your PR partly resolves the bug, but I see a few issues with it.

1) I dont understand why we are excluding 'post' here.
2) $post_type is always going to be a string in this scenario.

I would remove the if and use $type->name instead of the is_string line
what do you think?